docs: add `require.resolveWeak` module method (#10203)
web-infra-dev/rspack: website/docs/en/api/runtime-api/module-methods.mdx

### require.resolve

> Synchronously retrieve a module's ID. The compiler will ensure that the dependency is available in the output bundle. It is recommended to treat it as an opaque value which can only be used with `require.cache[id]` or `__webpack_require__(id)` (best to avoid such usage).

Reviewer: ```suggestion
Synchronously retrieves the module ID without executing the module code. This method will include the module in the final bundle.. The returned module ID is primarily intended for use with `require.cache[id]` or the Rspack internal `__webpack_require__(id)` function. Direct use of this method should be avoided in most application scenarios.
```

}
```

### require.resolveWeak

<ApiMeta specific={['Rspack', 'Webpack']} />

> Similar to `require.resolve`, but this won't pull the module into the bundle. It's what is considered a "weak" dependency.

Reviewer: ```suggestion
Similar to `require.resolve`, but this method won't pull the module into the final bundle. It's what is considered a "weak" dependency.
```

```js
require.resolveWeak(dependency: string);

> ```

Reviewer: Can we add the same example as webpack: https://webpack.js.org/api/module-methods/#requireresolveweak

require.resolveWeak(dependency: string);
```

> example:

Reviewer: ```suggestion
For example:
```
